src/test/java/kitchenpos/application/TableGroupServiceTest.java

```diff
@@ -0,0 +1,171 @@
+        final TableGroup tableGroup = new TableGroup(List.of(orderTable1, orderTable2));
+
+        // when & then
+        assertThatThrownBy(() -> tableGroupService.create(tableGroup))
+                .isInstanceOf(IllegalArgumentException.class);
+    }
+
+    @Test
+    void create_tableDuplicateException() {
+        // given
+        final OrderTable orderTable = tableService.create(new OrderTable(0, true));
+        final TableGroup tableGroup = new TableGroup(List.of(orderTable, orderTable));
+
+        // when & then
+        assertThatThrownBy(() -> tableGroupService.create(tableGroup))
+                .isInstanceOf(IllegalArgumentException.class);
+    }
+
+    @Test
+    void create_tableNotEmptyException() {
+        // given
+        final OrderTable orderTable1 = tableService.create(new OrderTable(0, true));
+        final OrderTable orderTable2 = tableService.create(new OrderTable(0, false));
+        final TableGroup tableGroup = new TableGroup(List.of(orderTable1, orderTable2));
+
+        // when & then
+        assertThatThrownBy(() -> tableGroupService.create(tableGroup))
+                .isInstanceOf(IllegalArgumentException.class);
+    }
+
+    @Test
+    void create_tableHasGroupException() {
+        // given
+        final OrderTable orderTable1 = tableService.create(new OrderTable(0, true));
+        final OrderTable orderTable2 = tableService.create(new OrderTable(0, true));
+        tableGroupService.create(new TableGroup(List.of(orderTable1, orderTable2)));
+
+        final OrderTable orderTable3 = tableService.create(new OrderTable(0, true));
+        final TableGroup tableGroup = new TableGroup(List.of(orderTable1, orderTable3));
+
+        // when & then
+        assertThatThrownBy(() -> tableGroupService.create(tableGroup))
+                .isInstanceOf(IllegalArgumentException.class);
+    }
+
+    @Test
+    void ungroup() {
+        // given
+        final OrderTable orderTable1 = tableService.create(new OrderTable(0, true));
+        final OrderTable orderTable2 = tableService.create(new OrderTable(0, true));
+        final TableGroup tableGroup = tableGroupService.create(new TableGroup(List.of(orderTable1, orderTable2)));
+
+        // when
+        tableGroupService.ungroup(tableGroup.getId());
+
+        // then
+        final List<OrderTable> orderTables = orderTableDao.findAllByTableGroupId(tableGroup.getId());
+        final OrderTable savedOrderTable1 = orderTableDao.findById(orderTable1.getId()).get();
+        final OrderTable savedOrderTable2 = orderTableDao.findById(orderTable2.getId()).get();
+        assertSoftly(softly -> {
+            softly.assertThat(orderTables).hasSize(0);
+            softly.assertThat(savedOrderTable1.getTableGroupId()).isNull();
+            softly.assertThat(savedOrderTable1.isEmpty()).isFalse();
+            softly.assertThat(savedOrderTable2.getTableGroupId()).isNull();
+            softly.assertThat(savedOrderTable2.isEmpty()).isFalse();
+        });
```

Comment:
Optional로 반환되는 OrderTable을 assertSoftly 내부에서 검증과 함께 진행하는 것은 어떠신가요?

```suggestion
        // then
        final List<OrderTable> orderTables = orderTableDao.findAllByTableGroupId(tableGroup.getId());
        final Optional<OrderTable> maybeOrderTable1 = orderTableDao.findById(savedOrderTable1.getId());
        final Optional<OrderTable> maybeOrderTable2 = orderTableDao.findById(savedOrderTable2.getId());
        assertSoftly(softly -> {
            softly.assertThat(orderTables).hasSize(0);
            softly.assertThat(maybeOrderTable1).isPresent();
            softly.assertThat(maybeOrderTable2).isPresent();
            final OrderTable actualOrderTable1 = maybeOrderTable1.get();
            final OrderTable actualOrderTable2 = maybeOrderTable2.get();
            
            softly.assertThat(actualOrderTable1.getTableGroupId()).isNull();
            softly.assertThat(actualOrderTable1.isEmpty()).isFalse();
            softly.assertThat(actualOrderTable2.getTableGroupId()).isNull();
            softly.assertThat(actualOrderTable2.isEmpty()).isFalse();
        });
```

Reply:
assertSoftly 안에서는 테스트만 수행해야 하는 줄 알았는데, 생각해보니 당연히 안에서 테스트 이외의 동작도 실행할 수 있군요!
덕분에 새로운 방식을 깨달았습니다. 감사합니다.

Reply:
[ae70a76](https://github.com/woowacourse/jwp-refactoring/pull/490/commits/ae70a76459af678bb5b1a8b7affe94e7402f9932)

적용 완!